pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -84,10 +87,25 @@ public class SystemTopicBasedTopicPoliciesService implements TopicPoliciesServic
     @VisibleForTesting
     final Map<TopicName, List<TopicPolicyListener<TopicPolicies>>> listeners = new ConcurrentHashMap<>();
 
+    private final AsyncLoadingCache<NamespaceName, SystemTopicClient.Writer<PulsarEvent>> writerCache;
```

Comment:
Should we close this writer when calling `pulsar-admin namespace delete <NS>`? And maybe we should add these two tests:
- The writer will be removed when the namespace is deleted.
- The writer will not prevent a namespace deletion without `--force`.

Reply:
addressed.